src/gui/FileBrowser.cpp

```diff
@@ -642,6 +673,7 @@ void FileBrowserTreeWidget::contextMenuEvent(QContextMenuEvent* e)
 	{
 	case TypeFileItem: {
 		auto file = dynamic_cast<FileItem*>(item);
+		bool _isFavorite = isFavorite(file->fullName());
```

Comment:
I would prefer this:
You probably don't even need to do this though, just call it directly (simpler IMO).
```suggestion
		const auto isFavoriteItem = isFavorite(file->fullName());
```

Reply:
calling it directly is simpler, but I don't like the idea of running the same loop twice

Reply:
I think there is more refactoring that can be done here actually. I will make a new suggestion for you.

Reply:
The fact that we have to duplicate the "Add favorite", "Remove favorite" code here is even more reason why we need a class that serves as a common abstraction for `Directory` and `FileItem`. Maybe called `FileBrowserItem`, which would have a public, pure virtual member function `fullName`. Then we can apply that same code to both cases without this duplication.

For the code itself, I would discard the use of the two ternary operators to make things more clear. For example, you could write this like:

```cpp
if (isFavorite)
{
contextMenu.addAction(tr("Add favorite"), [&, this] { addFavorite(item->fullName() });
}
else
{
contextMenu.addAction(tr("Remove favorite"), [&, this] { removeFavorite(item->fullName() });
}
```

(The formatting needs to be fixed since I cannot figure out how to tab in GitHub Markdown).

Reply:
If you are by any chance wondering when to use a ternary vs an if-else branch, IMO it depends on readability: if you have to sacrifice readability its not worth it, but for simple ternary's that aren't nested (as it is here), most of the time it's fine.